src/qat/model/device.py

```diff
@@ -1,118 +1,272 @@
 from __future__ import annotations
 
+import uuid
 from typing import Optional
 
-from pydantic import Field
+import numpy as np
+from pydantic import Field, model_validator
+from pydantic_core import core_schema
 
-from qat.model.component import Component
-from qat.model.serialisation import Ref, RefDict, RefList
-from qat.purr.compiler.devices import ChannelType
+from qat.utils.pydantic import WarnOnExtraFieldsModel
+
+QubitId = int
```

Comment:
minor: can be hoisted up and put it somewhere common to all hw model stuff, to void specifying types all over ...